Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/TrainControlSystem.cs

```diff
@@ -130,6 +130,24 @@ public MonitoringDevice(MonitoringDevice other)
         MonitoringDevice EmergencyStopMonitor;
         MonitoringDevice AWSMonitor;
 
+        private bool simulatorEmergencyBraking = false;
+        public bool SimulatorEmergencyBraking {
+            get
+            {
+                return simulatorEmergencyBraking;
+            }
+            protected set
+            {
+                simulatorEmergencyBraking = value;
+
+                if (Script != null)
+#pragma warning disable CS0618 // SetEmergency is obsolete
+                    Script.SetEmergency(value);
+#pragma warning restore CS0618 // SetEmergency is obsolete
+                else
+                    Locomotive.TrainBrakeController.TCSEmergencyBraking = value;
```

Comment:
So existing scripts that do not implement HandleEvent() will no longer be notified of simulator emergency brake applications. Have we thought through all of the implications? Should we continue to support SetEmergency()?

Reply:
This is why you have deprecation information on the SetEmergency function. To give time for the developer to adapt their scripts before breaking compatibility. The pragma is only here to avoid triggering a warning at each compilation, but the developers will see the warning in their IDE for their scripts.

Reply:
Ah, I misread that. The pragma statements produce warnings, but do not disable the code. Thanks for clarifying.